source/slang/slang-ir.cpp

```diff
@@ -2081,7 +2081,32 @@ namespace Slang
         memset(&keyInst, 0, sizeof(keyInst));
         keyInst.m_op = kIROp_IntLit;
         keyInst.typeUse.usedValue = type;
-        keyInst.value.intVal = inValue;
+        // Truncate the input value based on `type`.
+        switch (type->getOp())
+        {
+        case kIROp_Int8Type:
+            keyInst.value.intVal = static_cast<int8_t>(inValue);
+            break;
+        case kIROp_Int16Type:
+            keyInst.value.intVal = static_cast<int16_t>(inValue);
+            break;
+        case kIROp_IntType:
+            keyInst.value.intVal = static_cast<int32_t>(inValue);
+            break;
+        case kIROp_UInt8Type:
+            keyInst.value.intVal = static_cast<uint8_t>(inValue);
+            break;
+        case kIROp_UInt16Type:
+            keyInst.value.intVal = static_cast<uint16_t>(inValue);
+            break;
+        case kIROp_BoolType:
```

Comment:
On bool shouldn't it do something like IRIntegerValue(inValue != 0) to ensure it's only 1 or 0?

Reply:
Bool on many platforms can have arbitrary values so the compiler probably shouldn't alter the value here.

Reply:
Hmm, not sure about this. This is the Slang IR - if code is going to process Slang IR it's typically not going to be able able to handle or it be desirable for *arbitrary* values for bools. It seems that part of the point of this bit of code is to convert values into valid values for the type.

Subsequent stages of IR for handling for targets such as emitting, converting the type or outputting to a target encoding seem more appropriate.